### PR TITLE
Added logic for displaying Compete button for active online participants

### PR DESCRIPTION
**Pull request by gogo4ds**
Compete button is hidden for everyone after Contest time ends, but there are still active participants
Closes https://github.com/SoftUni-Internal/suls-issues/issues/3758

### CODE REVIEW

File: Open Judge System/OJS.Services/Business/Contests/ContestsBusinessService.cs [hunk 1]
```diff
@@ -15,5 +16,34 @@ public class ContestsBusinessService : IContestsBusinessService
             this.contestsData
                 .GetByIdQuery(contestId)
                 .Any(c => !c.AllowedIps.Any() || c.AllowedIps.Any(ai => ai.Ip.Value == ip));
+
+        public bool CanUserCompeteByContestUserAndIsAdminOrLecturer(Contest contest, string userId, bool isAdminOrLecturer)
```
Comment (YanaSlavcheva) [inline Open Judge System/OJS.Services/Business/Contests/ContestsBusinessService.cs]: not good idea to have so complex object as parameter. Better pass `contestId` as parameter IMO

File: Open Judge System/OJS.Services/Business/Contests/ContestsBusinessService.cs [hunk 2]
```diff
@@ -15,5 +16,34 @@ public class ContestsBusinessService : IContestsBusinessService
...
+                    p.ContestEndTime > DateTime.Now);
+            }
+
+            return false;
+        }
+
+        public bool CanUserCompeteByContestUserAndIsAdminOrLecturer(int contestId, string userId, bool isAdminOrLecturer)
+        {
+            var contest = this.contestsData.GetById(contestId);
+            return this.CanUserCompeteByContestUserAndIsAdminOrLecturer(contest, userId, isAdminOrLecturer);
```
Comment (YanaSlavcheva) [inline Open Judge System/OJS.Services/Business/Contests/ContestsBusinessService.cs]: why do we have this other method? do we use it elsewhere?

File: Open Judge System/OJS.Services/Business/Contests/ContestsBusinessService.cs [hunk 3]
```diff
@@ -15,5 +16,41 @@ public class ContestsBusinessService : IContestsBusinessService
             this.contestsData
                 .GetByIdQuery(contestId)
                 .Any(c => !c.AllowedIps.Any() || c.AllowedIps.Any(ai => ai.Ip.Value == ip));
+
+        public bool CanUserCompeteByContestUserAndIsAdmin(
```
Comment (Innos) [inline Open Judge System/OJS.Services/Business/Contests/ContestsBusinessService.cs]: CanUserCompeteByContestByUserAndIsAdmin

File: Open Judge System/OJS.Services/Business/Contests/IContestsBusinessService.cs [hunk 4]
```diff
@@ -5,5 +5,18 @@
     public interface IContestsBusinessService : IService
     {
         bool IsContestIpValidByIdAndIp(int contestId, string ip);
```
Comment (Innos) [inline Open Judge System/OJS.Services/Business/Contests/IContestsBusinessService.cs]: IsContestIpValidByContesAndIp

File: Open Judge System/OJS.Services/Business/Contests/IContestsBusinessService.cs [hunk 5]
```diff
@@ -5,5 +5,18 @@
...
+
+        /// <summary>
+        /// Determines if a user can compete in a contest, depending of his role and the contest type
+        /// </summary>
+        /// <param name="contestId">The id of the contest</param>
+        /// <param name="userId">The id of the user</param>
+        /// <param name="isAdmin">Is the user administrator in the system</param>
+        /// <param name="allowToAdminAlways">If true, and the user is admin he will always be able to compete</param>
+        bool CanUserCompeteByContestUserAndIsAdmin(
```
Comment (Innos) [inline Open Judge System/OJS.Services/Business/Contests/IContestsBusinessService.cs]: CanUserCompeteByContestByUserAndIsAdmin

File: Open Judge System/OJS.Services/Business/Participants/IParticipantsBusinessService.cs [hunk 6]
```diff
@@ -7,8 +7,6 @@
 
     public interface IParticipantsBusinessService : IService
     {
-        bool CanCompeteByContestAndUserId(Contest contest, string userId);
-
         Participant CreateNewByContestUserIdIsOfficialAndIsAdmin(
```
Comment (Innos) [inline Open Judge System/OJS.Services/Business/Participants/IParticipantsBusinessService.cs]: CreateNewByContestByUserByIsOfficialAndIsAdmin

File: Open Judge System/OJS.Services/Business/Participants/ParticipantsBusinessService.cs [hunk 7]
```diff
@@ -21,21 +21,6 @@ public class ParticipantsBusinessService : IParticipantsBusinessService
...
-                    userId,
-                    contest.Id);
-
-                return contestEndTime.HasValue && contestEndTime >= DateTime.Now;
-            }
-
-            return contest.CanBeCompeted;
-        }
-
         public Participant CreateNewByContestUserIdIsOfficialAndIsAdmin(
```
Comment (Innos) [inline Open Judge System/OJS.Services/Business/Participants/ParticipantsBusinessService.cs]: CreateNewByContestByUserByIsOfficialAndIsAdmin

File: Open Judge System/OJS.Services/Data/Contests/ContestsDataService.cs [hunk 8]
```diff
@@ -62,5 +70,10 @@ public bool IsActiveById(int contestId)
                 .Where(c => c.Id == contestId)
                 .Any(c => c.Lecturers.Any(l => l.LecturerId == userId) ||
                     c.Category.Lecturers.Any(l => l.LecturerId == userId));
+
+        public bool IsUserParticipantInByContestAndUserId(int contestId, string userId) =>
```
Comment (Innos) [inline Open Judge System/OJS.Services/Data/Contests/ContestsDataService.cs]: IsUserParticipantInByContestAndUser

File: Open Judge System/OJS.Services/Data/Contests/IContestsDataService.cs [hunk 9]
```diff
@@ -28,5 +30,7 @@ public interface IContestsDataService : IService
         bool IsActiveById(int contestId);
 
         bool IsUserLecturerInByContestIdAndUserId(int contestId, string userId);
+
+        bool IsUserParticipantInByContestAndUserId(int contestId, string userId);
```
Comment (Innos) [inline Open Judge System/OJS.Services/Data/Contests/IContestsDataService.cs]: IsUserParticipantInByContestAndUser

**Pull request merged**